### PR TITLE
Render BMO deployment

### PR DESCRIPTION
**Pull request by kashifest**
This PR adds a render of BMO deployment. 

It takes care of the task done at #667

### CODE REVIEW

File: config/render/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,835 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  labels:
+    control-plane: controller-manager
```
Comment (maelk) [inline config/render/deployment.yaml:5]: This namespace is wrong, can you please change it in config/manager/manager.yaml ?

**Review by maelk — CHANGES_REQUESTED** [review]
Please add the rendered file in the hack/generate.sh file so that it is checked by the job

**Comment by kashifest** [comment]
> Please add the rendered file in the hack/generate.sh file so that it is checked by the job

It is already checked in generate.sh the whole co fig dir.

**Comment by maelk** [comment]
> It is already checked in generate.sh the whole config dir.

oh true! sorry

**Review by digambar15 — COMMENTED** [review]
Otherwise looks good

File: config/render/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,835 @@
...
+  name: system
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
```
[the remark this answers is not in the document]
Comment (kashifest) [reply config/render/deployment.yaml:10]: Didn't get you, do you suggest a change here?

**Review by maelk — APPROVED** [review]

File: config/render/deployment.yaml (hunk 1 above)
Comment (kashifest) [reply config/render/deployment.yaml:5]: Done

File: config/render/deployment.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (digambar15) [reply config/render/deployment.yaml:10]: No problem. That's fine. I can see change in manager.yaml.

**Review by digambar15 — APPROVED** [review]

**Comment by maelk** [comment]
Getting this in to unlock work on CAPM3 side. The namespace change is required because the recent version of Kustomize does not apply the "namespace" kustomization to the namespace object name itself.

/lgtm

If there are any comments, we will address them in a second PR.

**Comment by dhellmann** [comment]
This looks fine, but for my own education can you confirm that we're checking this rendered file in because CI jobs use it?

**Comment by kashifest** [comment]
Yes, this rendered file is here to make sure that duplication of name of the objects doesn't collide while we kustomize all resources in CI

**Comment by dhellmann** [comment]
> Yes, this rendered file is here to make sure that duplication of name of the objects doesn't collide while we kustomize all resources in CI

Great, thank you for explaining that!

**Comment by maelk** [comment]
It's actually not limited to CI. Whenever we run the kustomization in CAPM3, we are now using this file.

**Pull request merged**